src/components/Button/Button.tsx

```diff
@@ -87,51 +53,22 @@ export const Button: React.FC<Props> = ({
   variant,
   size,
   disabled,
-  fullWidth,
-  iconName,
-  iconPosition = 'before',
-  inverted,
-  loading,
   onClick,
-  screenReaderText,
   href,
   text,
   type,
-  hideText,
   ...other
 }) => {
-  const componentClassName = clsx(styles['button'], className, {
-    [styles['button--primary']]: variant === 'primary',
-    [styles['button--bare']]: variant === 'bare',
-    [styles['button--link']]: variant === 'link',
-    [styles['button--sm']]: size === 'sm',
-    [styles['button--table-header']]: variant === 'table-header',
-    [styles['button--inverted']]: inverted === true,
-    [styles['button--full-width']]: fullWidth,
-    [styles['eds-is-loading']]: loading,
-  });
-  const TagName = href ? 'a' : 'button';
-
-  const computedIcon = (
-    <>
-      {loading && (
-        <Icon
-          aria-hidden="true"
-          focusable={false}
-          name="spinner"
-          className={styles['button__icon']}
-        />
-      )}
-      {!loading && iconName && (
-        <Icon
-          aria-hidden="true"
-          focusable={false}
-          name={iconName}
-          className={styles['button__icon']}
-        />
-      )}
-    </>
+  const componentClassName = clsx(
+    styles['button'],
+    className,
+    variant === 'primary' && styles['button--primary'],
+    variant === 'bare' && styles['button--bare'],
+    variant === 'link' && styles['button--link'],
+    size === 'sm' && styles['button--sm'],
+    variant === 'table-header' && styles['button--table-header'],
   );
+  const TagName = href ? 'a' : 'button';
```

Comment:
This is not related to clearing out the props, but this paradigm is very different than what we had. We chose to split the clickable components out into `Button` and `Link` so it was clearer what the expected functionality of the component was for developers. The downside is then you have the complexity around sharing styles, and it was also very confusing for designers since the distinction between these components came down to the tag and functionality, not the visual appearance. I'm assuming we want to keep using separate components, but if people want to collapse them, I'm personally okay with that.

Reply:
Ah, the age old conflict.

We should stay with our current thinking - separate components. `Button` renders a button, and `Link` renders an `a`.

Reply:
Age-old conflict indeed! Just going to share a few things as food for thought: (@ifrost1 pinging you here too for visibility).

Here's how we've historically handled things:
- We handle `button`s that need to look like hyperlinks using `<Button variant="link">`
- Our `Link` component specifically deals with rendering an `a` tag that looks like a hyperlink. We don't tend to handle `a` tags that need to stylistically look like buttons using this component as it requires a bigger lift to represent all the button styles into the component
- As you've seen, we've typically used the `<Button href="#">` approach to swap out the appropriate HTML tag to handle button-looking links. As devs, we know the difference between a `button` and `a` is strictly behavioral, so we've taken that work on at the `Button` level. The `<Button href="#">` convention fits with a non-dev vernacular: "We need a primary button labeled "Shop new arrivals" that takes users to the new arrivals landing page." Hopefully that provides some color to the convention.

Given that CZI's culture is more savvy, I think it's perfectly fine to expect user devs to have the know-how to reach for a `Link` if they need an `a` tag and a `Button` if they need a `button`.

If not going the `Button` tagName switch route, there are two ways of handling sharing the styles between these two components:
1. Preserve the `ClickableStyle` component to manage the shared styles
2. Define the shared styles between `Link` and `Button` as mixins and instantiate those mixins in their respective components.

Either is perfectly valid. However, if going the `ClickableStyle` route, we'd recommend storing `ClickableStyle` in a directory like `components/utils` to differentiate them from user-facing UI components. It can get a bit confusing and messy to have helper/utility/pass-through components intermingled with user-facing UI components, so giving those components their own space helps keep things tidy.

I could make a case for going the mixin route: sharing clusters of styles is specifically what mixins do, and we tend to try to minimize abstractions and keep things as direct as possible (e.g. editing `Button` styles should intuitively involve editing `Button.module.css`). But honestly either approach is perfectly fine.

Sorry for the big ol' text wall to effectively say "all these approaches are perfectly fine". But it's good to hold stuff like this up to the light and hit it from all angles!

Reply:
This is great, thanks! I really appreciate the layout of the different options (and the different options they lead to, lol).

Based on how our variants work (with the multiplier of the `color` prop), I'm leaning really heavily towards keeping `ClickableStyle`, but we can discuss that in a PR with the actual styles in it so I don't have to try to describe the situation.